nilearn/decoding/decoder.py

```diff
@@ -387,6 +388,15 @@ def __init__(self, estimator='svc', mask=None, cv=10, param_grid=None,
         self.n_jobs = n_jobs
         self.verbose = verbose
 
+        # Determine the scoring strategy
+        if self.scoring is not None:
+            self._scorer = get_scorer(self.scoring)
+        else:
```

Comment:
should `fit` below use `self._scorer` instead of calling `check_scoring`? see:
```
 scorer = check_scoring(self.estimator, self.scoring)
```

although in any case it will not happen often that scoring is None, as users are not supposed to instantiate that class directly, but use Decoder or DecoderRegressor, that set defaults for `scoring`

Reply:
Good point, thanks! 
I updated the code to setup the scorer in `fit` rather than in the constructor, and made `scorer` an attribute of the decoder.

Reply:
looks great, thanks! one last detail is that I believe in scikit-learn's convention any attribute created during `fit` has a name ending with `_`, and the `__init__` does nothing besides storing the parameters passed to it. (I think it should be like this for `sklearn.base.clone` to work properly). So maybe it would stick more closely to sklearn's conventions if we did not initialize `scorer = None` in `__init__`, and added a `scorer_` attribute in `fit`?